cuda/components/prefix_sum.cu

```diff
@@ -49,15 +49,21 @@ template <typename IndexType>
 void prefix_sum(std::shared_ptr<const CudaExecutor> exec, IndexType *counts,
                 size_type num_entries)
 {
+    // prefix_sum should be on the valid array
+    GKO_ASSERT(num_entries > 0);
```

Comment:
How about just using
```suggestion
    if (num_entries > 0) {
```
This would make the behavior between cuda/hip and reference/omp match more closely, since there is nothing *inherently* wrong with computing the prefix sum of an empty array.